backend/msgpack/conn.go

```diff
@@ -148,22 +156,29 @@ func (c *connection) connectWithLock() error {
 	return nil
 }
 
-func (c *connection) checkReconnectWithLock() bool {
+func (c *connection) checkReconnectWithLock() error {
 	// If we haven't accumulated enough failures to warrant another reconnect
 	// and we haven't past the maximum duration since the last time we attempted
 	// to connect then we simply return false without reconnecting.
-	if c.numFailures <= c.threshold && c.nowFn().UnixNano() < c.lastConnectNanos+int64(c.maxDuration) {
-		return false
+	belowThreshold := c.numFailures <= c.threshold
```

Comment:
nit: take it or leave it, but feels easier to read if I reverse the logic to
```
tooManyFailures := c.numFailures > c.threshold
if !tooManyFailures && c.nowFn().UnixNano() < c.lastConnectNanos+int64(c.maxDuration) {
  return errNoActiveConnection
}
err := c.resetAndConnectWithLock()
if err == nil {
  c.threshold = c.initThreshold
  return nil
}
if tooManyFailures && c.threshold < c.maxThreshold {
  ...
}
return err
```

Reply:
+1